src/ansys/aedt/core/maxwell.py

```diff
@@ -1787,70 +1787,200 @@ def assign_radiation(self, assignment, radiation=None):
-        self.odesign.EnableHarmonicForceCalculation(
-            [
-                "EnabledObjects:=",
-                assignment,
-                "ForceType:=",
-                force_type,
-                "WindowFunctionType:=",
-                window_function,
-                "UseNumberOfLastCycles:=",
-                use_number_of_last_cycles,
-                "NumberOfLastCycles:=",
-                last_cycles_number,
-                "StartTime:=",
-                "0s",
-                "UseNumberOfCyclesForStopTime:=",
-                True,
-                "NumberOfCyclesForStopTime:=",
-                1,
-                "StopTime:=",
-                "0.01s",
-                "OutputFreqRangeType:=",
-                "Use All",
-                "CaculateForceType:=",
-                calculate_force + " Force",
-            ]
-        )
-        return True
+        if self.solution_type == "EddyCurrent":
+            if not self.is3d and axis != 2:
+                axis = 2
+                self.logger.warning("For 2D EddyCurrent solver only Z-axis is available.")
+            self.odesign.EnableHarmonicForceCalculation(
+                [
+                    "EnabledObjects:=",
+                    assignment,
+                    "Coordinate System:=",
+                    coordinate_system,
+                    "Axis:=",
+                    axis,
+                    "Is Positive:=",
+                    is_positive,
+                ]
+            )
+            return True
+        elif self.solution_type in ["Transient", "TransientAPhiFormulation"]:
+            force = ["Harmonic", "Transient", "Harmonic and Transient"]
+            calculate_force = force[calculate_force]
+            range_type = ["Use All", "Use Range", "Use Number"]
+            output_frequency_range_type = range_type[output_frequency_range_type]
+            if not self.is3d and self.geometry_mode == "about Z" or self.solution_type == "TransientAPhiFormulation":
+                if force_type == 0 and calculate_force == "Transient":
+                    calculate_force = "Harmonic"
+                    self.logger.warning(
+                        "Object-Based Transient Force calculation is not supported for "
+                        "non-rotational transient solutions. Only Harmonic Force will be calculated."
+                    )
+                if not self.is3d and self.geometry_mode == "about Z":
+                    if use_number_of_cycles_from_stop_time or use_number_of_cycles_for_stop_time:
+                        self.logger.warning(
+                            " ``number_of_cycles_from_stop_time´´ and ``number_of_cycles_for_stop_time´´"
+                            "are not available for TransientZ."
+                        )
+                        use_number_of_cycles_from_stop_time = False
+                        use_number_of_cycles_for_stop_time = False
```

Comment:
@anur7 this part is very confusing. I'd simplify it. I also add an extra check on 2d and transient because use_number_of_cycles_from_stop_time and use_number_of_cycles_for_stop_time can be True in 3d Transient and Transient Aphi designs.
But this part has to be reviewed.